lib/src/source/git.dart

```diff
@@ -571,33 +570,52 @@ class GitSource extends CachedSource {
   ///
   /// If [shallow] is true, creates a shallow clone that contains no history
   /// for the repository.
-  Future _clone(
+  Future<void> _clone(
     String from,
     String to, {
     bool mirror = false,
     bool shallow = false,
-  }) {
-    return Future.sync(() {
-      // Git on Windows does not seem to automatically create the destination
-      // directory.
-      ensureDir(to);
-      var args = [
+    String? ref,
+  }) async {
```

Comment:
I'm less sure about this.
Curious what @sigurdm thinks?

Maybe, we  should use [`git worktree`](https://git-scm.com/docs/git-worktree).

Also if we're going to change this logic, let's aim to make it compatible with git lfs.